cpp/perspective/src/cpp/computed_column_map.cpp

```diff
@@ -27,7 +37,14 @@ t_computed_column_map::remove_computed_columns(
     const std::vector<std::string>& names) {
     for (const auto& name : names) {
         if (m_computed_columns.count(name)) {
-            m_computed_columns.erase(name);
+            // Don't remove computed columns that are being referred to by
+            // multiple views.
+            if (m_computed_columns_count.at(name) > 1) {
+                m_computed_columns_count[name]--;
+            } else {
+                m_computed_columns.erase(name);
+                m_computed_columns_count.erase(name);
+            }
```

Comment:
The complexity this introduces may not be worth it IMO - e.g., another case that will trivially break is duplicate column name dependencies in separate viewers.  We've discussed removing intermediate columns entirely from engine `process()` - I propose this may be an easier approach at this point than properly modeling all caveats of the intermediate column dependency system as currently written.